src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2317,10 +2317,13 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnDnsResolveResult(otE
 
     error = FromOtDnsResponseToMdnsData(serviceInfo, type, dnsResult->mMdnsService, dnsResult->mServiceTxtEntry);
 
+    VerifyOrExit(error == CHIP_NO_ERROR, );
+
+    DeviceLayer::PlatformMgr().ScheduleWork(DispatchResolve, reinterpret_cast<intptr_t>(dnsResult));
+
 exit:
 
     dnsResult->error = error;
```

Comment:
Should move above the ScheduleWork() call.

Reply:
I ended up moving `DeviceLayer::PlatformMgr().ScheduleWork(DispatchResolve, reinterpret_cast<intptr_t>(dnsResult));` back to the exit block and instead returning from HandleNodeIdResolve on error.